lib/models/tags.coffee

```diff
@@ -0,0 +1,106 @@
+Promise = require('bluebird')
+isEmpty = require('lodash/isEmpty')
+errors = require('resin-errors')
+
+{
+	isId
+	findCallback
+	mergePineOptions
+} = require('../util')
+
```

Comment:
I actually originally placed this in utils but it then after some time it felt to "core" to be in utils since it felt as a base class factory for all tags as you said.
Yea I will try to put a decent comment on the top. I'm glad we get the same confusions & end up with similar solutions.